code/modules/lighting/light_effect_cast.dm

```diff
@@ -0,0 +1,943 @@
+			animate(alpha = initial(alpha), time = 0)
+		if (LIGHT_REGULAR_FLICKER)
+			animate(src, alpha = 180, time = 5, loop = -1, easing = CIRCULAR_EASING)
+			animate(alpha = 255, time = 5, loop = -1, easing = CIRCULAR_EASING)
+
+	var/list/cached_view = view(distance_to_wall_illum, src)
+	for (var/thing in cached_view)
+		if (ismob(thing))
+			var/mob/M = thing
+			M.check_dark_vision()
+		if (isturf(thing))
+			var/turf/T = thing
+			add_light_turf(T)
+			if (CHECK_OCCLUSION(T))
+				affected_shadow_walls += T
+
+				for (var/dir in cardinal)
+					var/turf/T2 = get_step(T, dir)
+					if (!(T2 in cached_view) && CHECK_OCCLUSION(T2))
+						affected_shadow_walls += T2
+
+	if(!isturf(loc))
+		for(var/turf/T in affecting_turfs)
+			T.lumcount = -1
+			T.luminosity = 0
+		affecting_turfs.Cut()
+		return
+
+/atom/movable/light/shadow/cast_light_init()
+	. = ..()
+
+	shadow_component_turfs = list()
+	for (var/stuff in shadow_component_atoms)
+		qdel(stuff)
+		shadow_component_atoms -= stuff
+
+	var/turf/last_turf_in_group = null
+	var/list/turf_group = list()
+	// View? It just works.
+	for (var/turf/T in spiral_block(get_turf(src), light_range, only_view = TRUE))
+
+		if (!CHECK_OCCLUSION(T))
+			continue
+
+		if (last_turf_in_group && ((get_dist(T, last_turf_in_group) > 1) || (turf_group.len >= TURF_GROUP_LENGTH)))
+			shadow_component_turfs += list(turf_group)
+			turf_group = list()
+
+		last_turf_in_group = T
+		turf_group += last_turf_in_group
+
+	shadow_component_turfs += list(turf_group)
+
+// ------- Adding turfs
+
+/atom/movable/light/proc/add_light_turf(var/turf/T)
+	// The luminosity cast over distant turfs thing is only going to be a problem for very, very big light sources
+	// These checks are here for optimisation and not do unnecessary works
+	if (light_range > 6 && get_dist(T, src) > 3)
+		T.lumcount = -1
+		T.luminosity = 1
+		T.light_sources += src
+		affecting_turfs += T
+
+/atom/movable/light/shadow/add_light_turf(var/turf/T)
+	return
+
+/atom/movable/light/proc/cull_light_turfs()
+	for (var/turf/T in affecting_turfs)
+		T.light_sources -= src
+		if (!(T.light_sources.len))
+			T.luminosity = 0
+	affecting_turfs = list()
+
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// -- The procs related to the source of light
+
+/atom/movable/light/proc/cast_main_light()
+
+	if(light_type == LIGHT_DIRECTIONAL)
+
+		icon = 'icons/lighting/directional_overlays.dmi'
+		light_range = 2.5
+
+	else
+		if (base_light_color_state == "white")
+
+
+		// An explicit call to file() is easily 1000 times as expensive than this construct, so... yeah.
+		// Setting icon explicitly allows us to use byond rsc instead of fetching the file everytime.
+		// The downside is, of course, that you need to cover all the cases in your switch.
+			switch (light_range)
+				if (1)
```

Comment:
same as elsewhere... wouldn't this be much simpler with a simple ternary?

Reply:
See the comment above: 		// Setting icon explicitly allows us to use byond rsc instead of fetching the file everytime.